README.md

```diff
@@ -465,13 +455,27 @@ withConnection(
 .fork(console.error, console.log)
 ```
 
-Be careful when cancelling a hooked Future. If the resource was acquired but not
-yet consumed, it will no longer be disposed. One way to work around this is to
-have the `consume` computation return a cancel function which forcefully
-disposes of the resource.
+In the case that a hooked Future is *cancelled* after the resource was acquired,
+`dispose` will be executed and immediately cancelled. This means that rejections
+which may happen during this disposal are **silently ignored**. To ensure that
+resources are disposed during cancellation, you might synchronously dispose
+resources in the `cancel` function of the disposal Future:
+
+```js
+const closeConnection = conn => Future((rej, res) => {
+
+  //We try to dispose gracefully.
+  conn.flushGracefully(err => {
+    if(err) return rej(err);
```

Comment:
Are you writing `return rej(err);` to save yourself from typing `{ rej(err); return; }`? If so, I urge you to reconsider as it makes the lambda's return value appear to be significant. I'd find this clearer:

```javascript
if (err == null) {
  conn.close();
  res();
} else {
  rej(err);
}
```